crates/wasi/src/preview2/ctx.rs

```diff
@@ -35,11 +35,10 @@ impl WasiCtxBuilder {
         // API.
         let insecure_random_seed =
             cap_rand::thread_rng(cap_rand::ambient_authority()).gen::<u128>();
-
         Self {
-            stdin: Box::new(pipe::ReadPipe::new(std::io::empty())),
-            stdout: Box::new(pipe::WritePipe::new(std::io::sink())),
-            stderr: Box::new(pipe::WritePipe::new(std::io::sink())),
+            stdin: Box::new(pipe::EmptyInputStream),
```

Comment:
Given the semantics of that this stream is never ready for reading, would a better default be the `ClosedInputStream`?